### ISSUE TITLE
uncaught exception: query function not defined for Select2 s2id_autogen1

### ISSUE THREAD

**Issue by dobromir-hristov** [tabacitu/dick#55]
This is because we are trying to init select2 two times on the same page. Happens on the Article Create page.
At this point, we are loading the select2.js file as many times as we have select2 elements on the page, along with its init call. This is causing problems.
Anyone have an idea for a fix?

**Comment by dobromir-hristov** [comment]
fields/assets/js/select2.blade.php
```js
$('.select2').each(function (i, obj) {
			if ($(obj).data("select2")== 'undefined')
			{
				$(obj).select2();
			}
		});
```

**Comment by tabacitu** [comment]
Hi, @dobromir-hristov ,

Thanks for the fix. It's now live on [dick/crud v0.8.17](https://github.com/tabacitu/crud/commit/e789f877fc67b13ed241a986fba0e17077e7f0a2).

Cheers!

**Comment by dobromir-hristov** [comment]
Select2 puts data on elements on init, why did you decide to put a dedicated "select2" attribute on the element? 
It surely works as well, just wondered if it has any benefits?

**Comment by tabacitu** [comment]
Hi,

The (== 'undefined') conditional did not work for me in Chrome for Mac, so I jumped the gun on a solution, which you are correct - was not optimal. Haste makes waste. So thank you for the gentle scolding, it's now fixed using the 'select2' attribute, with a different conditional, though, which I've tested and works on Chrome and Firefox.

`
$('.select2').each(function (i, obj) {
            if (!$(obj).data("select2"))
            {
                $(obj).select2();
            }
        });
`

https://github.com/tabacitu/crud/commit/a0fbae1b8a678bce339c09c455845848eaddfdf8

Cheers to you for letting me have it! :-)
